threads-updates (#919)
langchain-ai/docs: src/langsmith/online-evaluations.mdx

There are two types of online evaluations supported in LangSmith:

> - **[LLM-as-a-judge](/langsmith/evaluation-concepts#llm-as-judge)**: Use an LLM to evaluate your traces. Used as a scalable way to provide human-like judgement to your output (e.g. toxicity, hallucination, correctness, etc.). LLM-as-a-judge evaluators can be configured for two different levels of granularity:

Reviewer: ```suggestion
- **[LLM-as-a-judge](/langsmith/evaluation-concepts#llm-as-judge)**: Use an LLM to evaluate traces as a scalable substitute for human-like judgment (e.g., toxicity, hallucinations, correctness). Supports two different levels of granularity:
```

### Prerequisites

- Your tracing project must be using [threads](/langsmith/threads).
- The top-level inputs and outputs of each trace in a thread must have a `messages` key that contains a list of messages. We support messages in [LangChain](/langsmith/log-llm-trace#messages-format), [OpenAI Chat Completions](https://platform.openai.com/docs/api-reference/chat/create), and [Anthropic Messages](https://docs.claude.com/en/api/messages) formats.
    - If the top-level inputs and outputs of each trace only contain the latest message in the conversation, we will automatically combine messages across turns into a thread.
    - If the top-level inputs and outputs of each trace contain the full conversation history, we will use that directly.

<Note>
If your traces don't follow the format above, thread level evaluators won't work. You’ll need to update how you trace to LangSmith to ensure each trace’s top-level inputs and outputs contain a list of `messages`.

Please refer to the [troubleshooting](#troubleshooting) section for more information.
</Note>

### Configuration

1. Navigate to the **Tracing Projects** tab and select a tracing project.

> 2. Click **+ New** in the top right corner of the tracing project page >  **New Evaluator** > **Evaluate a multi-turn thread**

Reviewer: ```suggestion
2. Click **+ New** in the top right corner of the tracing project page >  **New Evaluator** > **Evaluate a multi-turn thread**.
```

> 3. **Name your evaluator**

Reviewer: Are these bolded instructions part of the UI text? If not, I would make plain text, and treat as a sentence. Leaving bold for the UI elements.

> 3. **Name your evaluator**

Author: bolded instructions are not part of the UI text in steps 3-8. If those are not bolded its hard visually to separate out the headline from the description of the action 

lmk if theres a better way

<Tip>
When first testing your evaluator, use a short idle time so you can see results quickly. Once validated, increase it to match your the expected length of user interactions.
</Tip>
6. **Configure your LLM-as-a-judge prompt.**<br />
Define what you want to evaluate. This prompt will be used to evaluate the thread. You can also configure which parts of the `messages` list are passed to the evaluator to control the content it receives:
    - All messages: Send the full message list.
    - Human and AI pairs: Send only user and assistant messages (excluding system messages, tool calls, etc.).
    - First human and last AI: Send only the first user message and the last assistant reply.

7. **Set up you feedback configuration**<br />
Configure a name for the feedback key, the format for the feedback you want to collect and optionally enable reasoning on the feedback.

<Warning>
We don't recommend using the same name for a thread-level evaluator and a run-level evaluator as it can be hard to distinguish between the two.
</Warning>

8. **Save your evaluator**

> Once you save your evaluator, your evaluator will be visible in the **Evaluators** tab. In order to test your evaluator, you will need to wait for idle time to pass for new threads that are created after you save your evaluator.

Reviewer: ```suggestion
After saving, your evaluator will appear in the **Evaluators** tab. You can test it once the idle time has passed for any new threads created after saving.
```

### Limits
These are the current limits for multi-turn online evaluators (subject to change). Please reach out if you are running into any of these limits.

- **Runs must be less than one week old**: When a thread becomes idle, only runs within the past 7 days are eligible for evaluation.
- **Maximum of 500 threads evaluated at once**: If you have more than 500 threads marked as idle in a five minute period, we will automatically sample beyond 500.

> - **Maximum of 100 multi-turn online evaluators per workspace**

Reviewer 2: currently the maximum is set at 10

### Troubleshooting

**Checking the status of your evaluator** <br />

> You can check when your evaluator was last run by heading to the **Evaluators** tab within a tracing project and clicking the Logs button for the evaluator you created view its run history.

Reviewer: ```suggestion
You can check when your evaluator was last run by heading to the **Evaluators** tab within a tracing project and clicking the **Logs** button for the evaluator you created view its run history.
```
